Documentation fixes (#500)
reserve-protocol/protocol: docs/system-design.md

- Token `{tok}`: A token that our protocol holds a balance of, mostly as backing for the RToken.

Some examples:

- In the USD+ RToken we have designed, the unit of account is USD. Among others, cUSDC is a collateral token with reference unit USDC and target unit USD, and aUSDP is a collateral token with reference token USDP and target unit USD.

- Let's say we're building a pure-stable USD basket, out of USDC, USDP, and DAI. The unit of account would surely be USD. Each collateral token would also be its own reference unit, and its target would be USD.

> - Perhaps we're interested in a USD-denominated basket of blue-chip cryptocurrencies.  This type of rToken could be a 50/50 basket of wstETH and yvwBTC, where the reference units could be ETH and wBTC, respectively.  The target units would then be ETH & BTC, while the `{UoA}` would be USD.  Thus, the *value* of the rToken would flucuate (according to its unit-of-account), but all other necessary properties could be maintained.

Reviewer: ```suggestion
- Perhaps we're interested in a USD-denominated basket of blue-chip cryptocurrencies.  This type of rToken could be a 50/50 basket of wstETH and yvwBTC, where the reference units could be ETH and wBTC, respectively.  The target units would then be ETH & BTC, while the `{UoA}` would be USD.  Thus, the *value* of the rToken would fluctuate (according to its unit-of-account), but all other necessary properties could be maintained.
```

Separate from these, a number in dimension `{BU}` ("basket units") is an amount of current baskets.

> Regarding `{UoA}` and `{target}`:
> While it will usually be the case that a collateral's `{target}` will be the same as its RToken's `{UoA}`, this is by no means a requirement.  The `{UoA}` is a way to value the RToken in terms of a single unit, while each collateral's `{target}` is the expected value of its liability, or its `{ref}`.  As in example #3 above, an RToken's collaterals may have completely different `{target}` units, but be valued by the same `{UoA}`.

Reviewer: ```suggestion
### Regarding `{UoA}` and `{target}`

While it will usually be the case that a collateral's `{target}` will be the same as its RToken's `{UoA}`, this is by no means a requirement.  The `{UoA}` is a way to value the RToken and its collateral in terms of a single unit, while each collateral's `{target}` is the expected value of its liability, or its `{ref}`.  As in example #3 above, an RToken's collaterals may have completely different `{target}` units, but be valued by the same `{UoA}`.
```

- The SHORT_FREEZER role should be assigned to an address that might reasonably be expected to be the first to detect a bug in the code and can act quickly, and with some tolerance for false positives though less than in pausing. If a bug is detected, a short freeze can be triggered which will automatically expire if it is not renewed by LONG_FREEZER. The OWNER (governance) may also step in and unfreeze at anytime.
- The LONG_FREEZER role should be assigned to an address that will highly optimize for no false positives. It is much longer than the short freeze. It exists so that in the case of a zero-day exploit, governance can act before the system unfreezes and resumes functioning.

## System Auctions

> There are a few different types of trades that the Reserve Protocol makes, each with their own impetus and expected frequency.
> - collateral to collateral (basket is changed by governance, or collateral defaults) (**rare**)
> - collateral to RSR (reward / yield distribution to stRSR) (**daily/weekly**)
> - rsr to collateral (insurance provided by stRSR in the event of a collateral default) (**rare**)
> - reward tokens to collateral (token rewards turned into revenue) (**daily/weekly**)
> 
> Each of these trades takes place by way of auction.  There is currently only one auction method (Gnosis EasyAuction), but the Reserve Protocol supports the ability to expand to other methods.
> 
> A good explainer for how Gnosis auctions work can be found (on their github)[https://github.com/gnosis/ido-contracts].
> 

Reviewer: ```suggestion
The Reserve Protocol makes a few different types of trades:

- from collateral to RSR or RToken, in order to distribute collateral yields. These happen often.
- from reward tokens to RSR or RToken, in order to distribute tokens rewards from collateral. These also happen often.
- collateral to collateral, in order to change the distribution of collateral due to a basket change. Basket changes should be rare, happening only when governance changes the basket, or when some collateral token defaults.
- RSR to collateral, in order to recollateralize the protocol from stRSR insurance, after a basket change. These auctions should be even rarer, happening when there's a basket change and insufficient capital to achieve recollateralization without using insure.

Each type of trade can currently happen in only one way; the protocol launches a Gnosis EasyAuction. The Reserve Protocol is designed to make it easy to add other trading methods, but none others are currently supported.

A good explainer for how Gnosis auctions work can be found (on their github)[https://github.com/gnosis/ido-contracts].

```
